library/x509.c

```diff
@@ -35,6 +35,8 @@
 #include MBEDTLS_CONFIG_FILE
 #endif
 
+#include "mbedtls/error.h"
```

Comment:
We should add includes after the `_USE_C` ifdef, here and all other applicable places in this PR.

Reply:
We only need the error codes from the header, we are not calling any of the functions and don't need the implementation.

Reply:
It is our library style to include only what you need. If `MBEDTLS_X509_USE_C` is not defined, you don't need `error.h` included. Nor do you need `mbedtls/x509.h`, `mbedtls/asn1.h`, `mbedtls/oid.h`, `stdio.h`, `string.h`, and other headers which currently are only included if `MBEDTLS_X509_USE_C` is defined. Your include should be added alongside the likes of `mbedtls/oid.h` inclusion per library convention.